crates/polars-plan/src/dsl/function_expr/cat.rs

```diff
@@ -48,3 +64,42 @@ fn get_categories(s: &Column) -> PolarsResult<Column> {
     let arr = rev_map.get_categories().clone().boxed();
     Series::try_from((ca.name().clone(), arr)).map(Column::from)
 }
+
+/// Apply a function to the categories of a categorical column and re-broadcast the result back to
+/// to the array.
+fn apply_to_cats<F, T>(s: &Column, mut op: F) -> PolarsResult<Column>
+where
+    F: FnMut(&StringChunked) -> ChunkedArray<T>,
+    ChunkedArray<T>: IntoSeries,
+    T: PolarsDataType,
+{
+    let ca = s.categorical()?;
+    let (categories, phys) = match &**ca.get_rev_map() {
+        RevMapping::Local(c, _) => (c, ca.physical().cast(&IDX_DTYPE)?),
+        RevMapping::Global(physical_map, c, _) => {
+            // Map physical to its local representation for use with take() later.
+            let phys = ca
+                .physical()
+                .apply(|opt_v| opt_v.map(|v| *physical_map.get(&v).unwrap()));
+            let out = phys.cast(&IDX_DTYPE)?;
+            (c, out)
+        },
+    };
+
+    // Apply function to categories
+    let categories = StringChunked::with_chunk(PlSmallStr::EMPTY, categories.clone());
```

Comment:
This should take the name of `s`.

Reply:
https://github.com/pola-rs/polars/pull/20211/commits/2b9edc38714261cffa02363479c6df502d14232f